specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/compute.json

```diff
@@ -0,0 +1,10921 @@
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyUpdateResource"
+            },
+            "description": "Parameters supplied to update the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyResource"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "SshPublicKeys"
+        ],
+        "operationId": "SshPublicKeys_Delete",
+        "description": "Delete an SSH public key.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "sshPublicKeyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "SshPublicKeys"
+        ],
+        "operationId": "SshPublicKeys_Get",
+        "description": "Retrieves information about an SSH public key.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "sshPublicKeyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the SSH public key."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SshPublicKeyResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get an ssh public key.": {
+            "$ref": "./examples/GetAnSshPublicKey.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/sshPublicKeys/{sshPublicKeyName}/generateKeyPair": {
+      "post": {
```

Comment:
long-ruunning?

Reply:
[5:25 PM] Nathan Kuchta
    No, they are not long running and return immediately